src/main/java/gololang/FunctionReference.java

```diff
@@ -49,7 +49,11 @@ public FunctionReference(MethodHandle handle, String[] parameterNames) {
       throw new IllegalArgumentException("A method handle cannot be null");
     }
     this.handle = handle;
-    this.parameterNames = parameterNames;
+    if (parameterNames != null) {
```

Comment:
I'm unsure a defensive copy here is helpful. In regular Java code I would say yes, but here we have an array that is kept private, and the values come from the runtime bootstrap.

The case of `null` is good, but again, is it worth allocating an array?